### PR TITLE
15 cl add debug toggle

### LINKED ISSUE #15: Add global debug values toggle
It would be nice if we had a button somewhere in the UI that would allow us to toggle the visibility of the debug info that we're seeing. Using the pub/sub design pattern I think this should be fairly straightforward to implement:

- [x] Add a check box with a label beside it saying "display debug info" on the sidebar that displays when nothing on the diagram is selected. The button should be unchecked by default.
- [x] Add code so that when the button is pushed, it publishes something along the lines of `"debug/toggle"`
- [x] Add a `displayDebugValues` variable to the `Node,js` class, initialized to `false`.
- [x] When initializing the Node, add a call to `subscribe("debug/toggle", function() { ... })` where the function simply makes `displayDebugValues` take the opposite value of what it currently has
- [x] For the lines displaying the debug info, add an `if (displayDebugValues)` check

### PR DESCRIPTION
**Pull request by nlehnert1**
Still haven't made a PR template, sorry

## What does this PR accomplish?
Resolves #15 

## How does it do it?
Made use of the pub/sub pattern. Added a line in `Sidebar.js` to insert a new button on the "Edit" sidebar, and gave it an onclick event of publishing `"debug/toggle"`. Added a `self.displayDebug` field to `Node.js`, and added line subscribing the node instance to the `"debug/toggle"` topic, whereupon it inverts the value of `displayDebug`, redraws the node, and then refreshes the background.

## Tada
image
image

## Be good
- [x] I've tested this feature as rigorously as I can
- [x] I've confirmed no new bugs being introduced from my work

### CODE REVIEW

File: v1.1.1/js/Sidebar.js [hunk 1]
```diff
@@ -174,14 +174,15 @@ function Sidebar(loopy){
 		page.addComponent(new ComponentHTML({
 			html: ""+
 			
-			"<b style='font-size:1.4em'>LOOPY</b> (v2.0)<br>a tool for thinking in systems<br><br>"+
+			"<b style='font-size:1.4em'>LOOPY</b> (v1.1.1)<br>a tool for thinking in systems<br><br>"+
```
Comment (UriBombastic) [inline v1.1.1/js/Sidebar.js:177]: Good catch

File: v1.1.1/js/Node.js [hunk 2]
```diff
@@ -280,12 +286,13 @@ function Node(model, config){
 	
 		var roundedValue = Math.round(self.value * 100) / 100; // Temp variable solely to cleanly display value
 		
-		// resize value text
-		self.fillSelfSizingText(ctx, r, roundedValue, 40); // Display value slightly below label.
+		if(self.displayDebug) {
+			// resize value text
+			self.fillSelfSizingText(ctx, r, roundedValue, 40); // Display value slightly below label.
```
Comment (UriBombastic) [inline v1.1.1/js/Node.js:291]: Do we want the value to be lumped in with the debug text? I thought that the value display was something that we genuinely wanted for the users to be able to see.

**Review by UriBombastic — CHANGES_REQUESTED** [review]
Looks mostly good to me, I just have a few questions as addressed in comments.
Additionally, I can't quite figure out where this happens, but do we really want the radius to reset when we toggle the debugs? It doesn't even fully reset the simulation, leaving signals traveling through edges and exploded nodes as exploded, which I feel is unwanted behaviour.

File: v1.1.1/js/Node.js (hunk 2 above)
Comment (nlehnert1) [reply v1.1.1/js/Node.js:291]: We can maybe separate it out later on, but Erin mentioned that once the value was displayed, she couldn't help but like obsess over what the value was, and that's not the real focus of loopy. We may or may not make a separate toggle for it, but either way I think it sounds like we won't be having it on by default anymore.

File: v1.1.1/js/Sidebar.js (hunk 1 above)
Comment (nlehnert1) [reply v1.1.1/js/Sidebar.js:177]: ty ty

**Review by UriBombastic — APPROVED** [review]
LGTM

**Pull request merged**